code/core/src/main/java/com/adobe/marketing/mobile/internal/util/ActivityCompatOwner.kt

```diff
@@ -0,0 +1,152 @@
+ * from the SDK is being attached.
+ * This is required to provide a lifecycle owner to the view tree of an activity that inherits
+ * from android.app.Activity which does not provide a lifecycle owner by default. Not doing so will
+ * result in a crash when trying to attach a compose view. Android recommends using inheriting from
+ * AppCompatActivity or similar which provides a lifecycle owner by default (irrespective of whether
+ * compose is being used or not). This is a best effort to provide a lifecycle owner when this is not
+ * the case.
+ */
+internal class ActivityCompatOwner :
+    LifecycleOwner,
+    ViewModelStoreOwner,
+    SavedStateRegistryOwner,
+    OnBackPressedDispatcherOwner {
+
+    // LifecycleOwner methods
+    private val lifecycleRegistry: LifecycleRegistry = LifecycleRegistry(this)
+    override val lifecycle: Lifecycle
+        get() = lifecycleRegistry
+
+    // ViewModelStore methods
+    private val store = ViewModelStore()
+    override val viewModelStore: ViewModelStore
+        get() = store
+
+    // SavedStateRegistry methods
+    private val savedStateRegistryController = SavedStateRegistryController.create(this)
+    override val savedStateRegistry: SavedStateRegistry
+        get() = savedStateRegistryController.savedStateRegistry
+
+    // OnBackPressedDispatcherOwner methods
+    private val dispatcher = OnBackPressedDispatcher {}
+    override val onBackPressedDispatcher: OnBackPressedDispatcher
+        get() = dispatcher
+
+    /**
+     * Trigger the ON_CREATE lifecycle event for this [ActivityCompatOwner].
+     */
+    internal fun onCreate() {
+        savedStateRegistryController.performRestore(null)
+        lifecycleRegistry.handleLifecycleEvent(Lifecycle.Event.ON_CREATE)
+    }
+
+    /**
+     * Trigger the ON_DESTROY lifecycle event for this [ActivityCompatOwner].
+     */
+    internal fun onDestroy() {
+        lifecycleRegistry.handleLifecycleEvent(Lifecycle.Event.ON_DESTROY)
+        store.clear()
+    }
+
+    /**
+     * Attaches this [ActivityCompatOwner] to the given view.
+     * @param view the view to attach this [ActivityCompatOwner] to
+     */
+    internal fun attachToView(view: View?) {
+        view?.apply {
+            setViewTreeLifecycleOwner(this@ActivityCompatOwner)
+            setViewTreeViewModelStoreOwner(this@ActivityCompatOwner)
+            setViewTreeSavedStateRegistryOwner(this@ActivityCompatOwner)
+            setViewTreeOnBackPressedDispatcherOwner(this@ActivityCompatOwner)
+        }
+    }
+
+    /**
+     * Detaches this [ActivityCompatOwner] from the given view.
+     * @param view the view to detach this [ActivityCompatOwner] from
+     */
+    internal fun detachFromView(view: View?) {
+        view?.apply {
+            setViewTreeLifecycleOwner(null)
+            setViewTreeViewModelStoreOwner(null)
+            setViewTreeSavedStateRegistryOwner(null)
```

Comment:
Did you miss clearing `setViewTreeOnBackPressedDispatcherOwner`?

Reply:
The backPressedDispatcherOwner will be cleared automatically. The`setViewTreeOnBackPressedDispatcherOwner` does not accept null's.